client/host_queue.go

```diff
@@ -169,46 +169,62 @@ func (q *queue) rotateOpsWithLock() []op {
 }
 
 func (q *queue) drain() {
-	wgAll := &sync.WaitGroup{}
+	var (
+		wgAll                       = &sync.WaitGroup{}
+		currWriteOpsByNamespace     = make(map[string][]op)
```

Comment:
There is a new integration test that covers scenarios including fetching from different keyspaces etc. I can add another unit test for this.